test/test-router.js

```diff
@@ -0,0 +1,152 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const faker = require('faker');
+const mongoose = require('mongoose');
+
+const should = chai.should();
+chai.use(chaiHttp);
+const {User} = require('../routes/models');
+const {app, runServer, closeServer} = require('../server');
+const {TEST_DATABASE_URL} = require('../config/database');
+
+
+function seedUserData() {
+	console.log('seeding user data');
+	const seedData = [];
+
+	for (let i=1; i<=10;i++) {
+		seedData.push(generateUserData());
+	}
+
+	return User.insertMany(seedData);
+}
+
+function generateUserData() {
+	return {
+		username: faker.internet.userName(),
+		firstName: faker.name.firstName(),
+		lastName: faker.name.lastName(),
+		password: faker.internet.password()
+	}
+}
+
+function tearDownDb() {
+	console.log('Deleting database');
+	return mongoose.connection.dropDatabase();
+}
+
+describe('Users source', function() {
+
+	before(function() {
+		return runServer(TEST_DATABASE_URL);
+	});
+
+	beforeEach(function() {
+		return seedUserData();
+	})
+
+	afterEach(function() {
+		return tearDownDb();
+	})
+
+	after(function() {
+		return closeServer();
+	});
+
+	describe('GET root endpoint', function() {
+
+		it('should return status 200 on GET', function() {
+
+			return chai.request(app)
+			.get('/')
+			console.log('got')
+			.then(function(res) {
+				res.should.have.status(200);
+			});
+		});
+	});
+
+	describe('POST endpoint', function() {
+
+		it('should create new user account', function() {
+			const newUser = generateUserData();
+			return chai.request(app)
+			.post('/signup')
+			.send(newUser)
+			.then(function(res) {
+				res.should.have.status(201);
+				res.should.be.json;
+				res.body.should.be.a('object');
+				res.body.should.include.keys(
+					'id',
```

Comment:
Like this? 
`res.body.should.include.keys('id','username',	'firstName','lastName');`
Or are you talking about lines 78 through 81, all of it.  If so, why is it so important?